lib/typing/parse.go

```diff
@@ -0,0 +1,90 @@
+package typing
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/artie-labs/transfer/lib/typing/decimal"
+	"github.com/artie-labs/transfer/lib/typing/ext"
+)
+
+func ParseValue(settings Settings, key string, optionalSchema map[string]KindDetails, val any) KindDetails {
```

Comment:
No functionality change apart from the fact that I split `ParseValue` into 2 separate functions